apps/omg_watcher_info/priv/repo/migrations/20200211064454_add_txtype_to_transaction_and_output.exs

```diff
@@ -0,0 +1,75 @@
+defmodule OMG.WatcherInfo.DB.Repo.Migrations.AddTxtypeToTransactionAndOutput do
+  use Ecto.Migration
+
+  import Ecto.Query, only: [from: 2]
+
+  alias Ecto.Adapters.SQL
+  alias OMG.State.Transaction
+  alias OMG.WatcherInfo.DB.Repo
+
+  def up() do
+    alter table(:transactions) do
+      add :txtype, :integer
+    end
+    alter table(:txoutputs) do
+      add :otype, :integer
+    end
+    create index(:transactions, :txtype)
+    create index(:txoutputs, :otype)
+    flush()
+
+    set_txtypes()
+
+    alter table(:transactions) do
+      modify(:txtype, :integer, null: false)
+    end
+    alter table(:txoutputs) do
+      modify(:otype, :integer, null: false)
+    end
+  end
+
+  def down() do
+    # This migration only supports outputs of type 1 and 2, we prevent rollback so we
+    # don't have problems if new types are introduced in the future.
+    raise "can't rollback this migration"
+  end
+
+  # Update existing transactions and output that don't have a type
+  defp set_txtypes() do
+    :ok =
+      Repo
+      |> SQL.query!("SELECT txhash, txbytes FROM transactions")
+      |> Map.get(:rows)
+      |> Enum.reduce(%{}, fn [txhash, txbytes], acc ->
+        %{raw_tx: %{tx_type: txtype}} = Transaction.Signed.decode!(txbytes)
+        hashes = [txhash | acc[txtype] || []]
```

Comment:
if `acc[txtype]` is nil use `[]`?

Reply:
Yeah agree here it took me some time to parse. Maybe split into 2 statements or `Map.update(acc, txtype, [txhash], fn hashes -> [txhash | hashes] end)`.